### PR TITLE
#1 Graph rendering

### PR DESCRIPTION
**Pull request by michaelKurowski**

### CODE REVIEW

**Comment by michaelKurowski** [comment]
TODO
Rendering graph from data instead of generating nodes during rendering

**Comment by michaelKurowski** [comment]
TODO
Laggs. Biggest source of laggs are captions and beziers that fade out on the border of rendering.

**Comment by michaelKurowski** [comment]
TODO
Fix navigating directly to third-level nodes

**Comment by michaelKurowski** [comment]
TODO
Fix edge beziers not always showing up

**Comment by Jordan141** [comment]
nice node modules

**Comment by michaelKurowski** [comment]
@Jordan141 ready for code review

**Comment by michaelKurowski** [comment]
Actually I'll refactor and optimize it first

**Comment by michaelKurowski** [comment]
> TODO
> Laggs. The biggest source of laggs are captions and beziers that fade out on the border of rendering.
> Beziers at the edge could be just rendered lower, under a layer that would be covered with radial gradient

Actually after hiding of edge beziers the laggs still occur on mobile phones

**Comment by michaelKurowski** [comment]
@Jordan141 could you check whether it's laggy on your phone?

**Comment by michaelKurowski** [comment]
TODO:
Sometimes when  we click in a node which has two emerging edge bezier lines, the transition malfunction.

**Comment by michaelKurowski** [comment]
TODO animation malfunction after navigating to emYhGb... and then to hUM9J...

image
image

**Comment by michaelKurowski** [comment]
Alright @Jordan141 done, please CR

File: pocs/dataRepresentation/public/index.js [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+            duration: MOVE_DURATION,
+            onFinish: cb
+        })
+    }
+    hideDedscendantNodes(rootNode, preserveNode)
+    if (!rootNode.parent) return
+    hideAncestorNodes(rootNode.parent, preserveNode)
+}
+
+function hideDedscendantNodes(rootNode, preserveNode) {
```
Comment (Jordan141) [inline pocs/dataRepresentation/public/index.js:70]: Spelling error, should be hideDescendentNodes

File: pocs/dataRepresentation/public/index.js [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+            node: childNode,
+            parentPosition: childPosition,
+            parentDegree: childDegree
+        })
+        layer.add(childNode.konva.caption)
+        
+        layer.add(childNode.konva.circle);
+        
+    }
+}
```
Comment (Jordan141) [inline pocs/dataRepresentation/public/index.js:300]: There's a lot going on in here, might be worth refactoring this

Comment (Jordan141) [reply pocs/dataRepresentation/public/index.js:300]: by here I mean the above function, drawLevel

File: pocs/dataRepresentation/public/index.js [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+        node.konva.bezier.to({
+            points: [...bezierData],
+            duration: MOVE_DURATION,
+            easing: Konva.Easings.EaseInOut
+        })
+    }
+}
+
+
+// MATH UTILITY FUNCTIONS
```
Comment (Jordan141) [inline pocs/dataRepresentation/public/index.js:518]: Utility functions should be brought out into a utility module

**Comment by Jordan141** [comment]
Non-MVP Requests:
- Right click should not perform the functionality of left click

------------

Other than that I will approve the PR

**Comment by Jordan141** [comment]
Bug found:
If you click on the next node before the animation ends, you become unable to return to the previous level.

**Pull request merged**